DEVDOCS-5033: [new] B2B getting started + GraphQL Storefront API auth narrative (#288)
bigcommerce/docs: docs/api-docs/headless/customers.mdx

---
title: Managing customers
keywords: customers, currency, currencies, headless, graphql, storefronts,  
---

# Managing customers

This section covers different ways to associate customers with headless carts.

## GraphQL Storefront API

You can use the GraphQL Storefront API to sign in to a customer account using hosted client-side, headless client-side, or server-side code. 

### Hosted client-side sign in

To sign in a customer account with an email address and a password in client-side code on a hosted storefront, use the [login mutation](/docs/start/authentication/graphql-storefront#signing-customers-in).

> When a customer is signed in using the customer login mutation, subsequent queries made to the GraphQL Storefront API will return customer-specific results, such as customer group pricing.

Author: ```suggestion
When you sign in a customer using the login mutation, subsequent queries made to the GraphQL Storefront API will return customer-specific results, such as customer group pricing.
```

### Headless and server-side sign in

To make queries from the perspective of a particular customer in headless or server-side code, use [customer impersonation tokens](/docs/start/authentication/graphql-storefront#customer-impersonation-tokens) rather than storefront tokens. 

> Depending on your implementation, you can use the login mutation or a combination of the [Current Customer API](/docs/start/authentication/current-customer), the [Customer Login API](/docs/start/authentication/customer-login), or the REST Management API's [customers feature](/docs/rest-management/customers) to authenticate the customer and get their customer ID. Pass the customer ID with your GraphQL Storefront requests using the `X-Bc-Customer-Id` header to interact with the store fron the customer's perspective. The responses will reflect customer-specific pricing, product availability, account information, and more.

Author: ```suggestion
Depending on your implementation, you can use the login mutation or a combination of the [Current Customer API](/docs/start/authentication/current-customer), the [Customer Login API](/docs/start/authentication/customer-login), or the REST Management API's [customers feature](/docs/rest-management/customers) to authenticate the customer and get their customer ID. Pass the customer ID with your GraphQL Storefront requests using the `X-Bc-Customer-Id` header to interact with the store from the customer's perspective. The responses will reflect customer-specific pricing, product availability, account information, and more.
```

## Customer single sign-on

> When a customer signs in to your headless storefront and your application redirects them to a BigCommerce-hosted page, such as checkout, you can use the [Customer Login API](/docs/start/authentication/customer-login) to create a single sign-on experience between your headless storefront and BigCommerce. Doing so will make the redirection from the headless storefront to BigCommerce a more seamless experience for the customer.

Author: ```suggestion
When a customer signs in to your headless storefront and your application redirects them to a BigCommerce-hosted page, such as checkout, you can use the [Customer Login API](/docs/start/authentication/customer-login) to sign them in to the hosted session. 
```
